lightning/src/ln/channelmanager.rs

```diff
@@ -1563,61 +1563,15 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
-	/// Call this upon creation of a funding transaction for the given channel.
-	///
-	/// Returns an [`APIError::APIMisuseError`] if the funding_transaction spent non-SegWit outputs
-	/// or if no output was found which matches the parameters in [`Event::FundingGenerationReady`].
-	///
-	/// Panics if a funding transaction has already been provided for this channel.
-	///
-	/// May panic if the output found in the funding transaction is duplicative with some other
-	/// channel (note that this should be trivially prevented by using unique funding transaction
-	/// keys per-channel).
-	///
-	/// Do NOT broadcast the funding transaction yourself. When we have safely received our
-	/// counterparty's signature the funding transaction will automatically be broadcast via the
-	/// [`BroadcasterInterface`] provided when this `ChannelManager` was constructed.
-	///
-	/// Note that this includes RBF or similar transaction replacement strategies - lightning does
-	/// not currently support replacing a funding transaction on an existing channel. Instead,
-	/// create a new channel with a conflicting funding transaction.
-	pub fn funding_transaction_generated(&self, temporary_channel_id: &[u8; 32], funding_transaction: Transaction) -> Result<(), APIError> {
-		let _persistence_guard = PersistenceNotifierGuard::new(&self.total_consistency_lock, &self.persistence_notifier);
-
-		for inp in funding_transaction.input.iter() {
-			if inp.witness.is_empty() {
-				return Err(APIError::APIMisuseError {
-					err: "Funding transaction must be fully signed and spend Segwit outputs".to_owned()
-				});
-			}
-		}
-
+	/// Handles the generation of a funding transaction, optionally (for tests) with a function
+	/// which checks the correctness of the funding transaction given the associated channel.
+	fn funding_transaction_generated_intern<Checker: Fn(&Channel<Signer>, &Transaction) -> Result<u16, APIError>>
```

Comment:
The checker interface may be more intuitive if it returned an `OutPoint` and perhaps named `find_funding_output`. Otherwise, it could use some documentation on the expected return value.